growcrops.lua

```diff
@@ -0,0 +1,138 @@
+-- Instantly grow crops in farm plots
+
+local args = {...}
+local material = args[1]
+local crop_raws = cacheCropRaws()
+local inventory = collectInventory(crop_raws)
+
+if not material or material == "help" or material == "list" then
+    listPlantedCrops(crop_raws, inventory)
+else
+    if material == 'all' then
+        for mat_idx, _ in pairs(inventory) do
+            growCrop(crop_raws, crop_raws[mat_idx].name)
+        end
+    else
+        growCrop(crop_raws, material)
+    end
+end
+
+-- Collects crop name and growdur from the raws.
+-- The returned table can be indexed by mat_index
+function cacheCropRaws()
+    local crops = {}
+    -- store ID and grow duration for each crop
+    for idx, p in ipairs(df.global.world.raws.plants.all) do
+        crops[idx] = { name = p.id, growdur = p.growdur }
+    end
+
+    return crops
+end
+
+-- determine if the given refs collection contains the type
+-- e.g. containsRefType(item.general_refs, df.general_ref_building_holderst)
+function containsRefType(refs, type)
+    for i, ref in ipairs(refs) do
+        if ref._type == type then
+            return true
+        end
+    end
+
+    return false
+end
```

Comment:
This seems like a common thing that scripts would want to do but my incredibly limited knowledge of lua and completely non-existant knowledge of the df api lead me to just implement this method on my own as I couldn't locate an alternative way to do it.

Is there another way?